corehq/ex-submodules/dimagi/utils/django/email.py

```diff
@@ -168,3 +169,75 @@ def send_HTML_email(subject, recipient, html_content, text_content=None,
                 bcc=bcc,
             )
             error_msg.send()
+
+
+def getEmailConfiguration(domain: str, from_email: str = settings.DEFAULT_FROM_EMAIL):
+    from corehq.apps.email.models import EmailSettings
+    try:
+        email_setting = EmailSettings.objects.get(domain=domain)
+        if email_setting.use_this_gateway:
+            return CustomEmailConfiguration(email_setting)
+        else:
+            return DefaultEmailConfiguration(from_email)
```

Comment:
```suggestion
        email_setting = EmailSettings.objects.get(domain=domain, use_this_gateway=True)
        return CustomEmailConfiguration(email_setting)
```

Reply:
Is that equivalent? email_setting would be empty if `use_this_gateway` is False and it wouldn't return `DefaultEmailConfiguration`

Reply:
It just moves the check to the DB lookup - if there's nothing on that domain with `use_this_gateway=True`, it'll raise `DoesNotExist` and return the default after the `except`

Reply:
you're right, i forgot i had that enclosed in a try block [9cf170c](https://github.com/dimagi/commcare-hq/pull/33484/commits/9cf170cd8c05e9c4a05696c221cf626172557c53)